src/deps.rs

```diff
@@ -0,0 +1,61 @@
+use bevy::asset::{HandleId, HandleUntyped};
+use bevy::utils::HashMap;
+use std::path::PathBuf;
+
+/// A map of assets depended upon by a [prototypical] asset.
+///
+/// Assets are broken into two categories: templates and general assets.
+/// Both are kept loaded as long as the prototype remains loaded.
+///
+/// [prototypical]: crate::Prototypical
+#[derive(Default, Debug, PartialEq)]
+pub struct DependencyMap {
+    /// A mapping of a template to a _strong_ handle of its asset.
+    templates: HashMap<PathBuf, HandleUntyped>,
```

Comment:
If these are themselves assets, we may want to key with bevy's `AssetPath` here instead of `PathBuf`.

Reply:
The only issue with `AssetPath` is that requires a lifetime (I think).